### PR TITLE
Add OpenStack metrics scaler doc

### PR DESCRIPTION
**Pull request by Rodolfodc**
Signed-off-by: rodolfodc <r.costa@sidi.org.br>

Add Openstack metrics scaler documenting (previously this scaler would be Openstack AODH scaler, but call it openstack metric scaler fits better its purpose.

This PR adds documentation for OpenStack Metric scaler.

PR for the scaler can be found at [kedacore/keda#1462](https://github.com/kedacore/keda/pull/1692).

Relates to [kedacore/keda#1342](https://github.com/kedacore/keda/issues/1382).

### CODE REVIEW

**Comment by tomkerkhove** [comment]
Is it me or is the diff empty?

**Comment by Rodolfodc** [comment]
You are right! I'm gonna check what happened.

File: content/docs/2.2/scalers/openstack-metric.md [hunk 1]
```diff
@@ -0,0 +1,174 @@
...
+draft: true
+---
++++
+title = "Openstack Metric"
+layout = "scaler"
+availability = "v2.2+"
+maintainer = "Community"
+description = "Scale applications based on the metrics collected from an specific resource."
+go_file = "openstack_metric_scaler"
++++
```
Comment (tomkerkhove) [inline content/docs/2.2/scalers/openstack-metric.md:13]: This doesn't look correct?

File: content/docs/2.2/scalers/openstack-metric.md [hunk 2]
```diff
@@ -0,0 +1,174 @@
+---
+title: "Openstack Metric"
+date: 2021-03-07T17:14:29-03:00
+draft: true
+---
++++
+title = "Openstack Metric"
+layout = "scaler"
+availability = "v2.2+"
```
Comment (tomkerkhove) [inline content/docs/2.2/scalers/openstack-metric.md:9]:
```suggestion
availability = "v2.3+"
```

File: content/docs/2.2/scalers/openstack-metric.md [hunk 3]
```diff
@@ -0,0 +1,174 @@
...
+
+- `authURL` - The Keystone authentication URL. The pattern is: `http://<host>:<keystone_port>/<keystone_version>/`
+- `appCredentialID` - The Application Credential ID
+- `appCredentialSecret` - The Application Credential secret
+
+### Example
+
+#### Password method
+
+Here is an example of how to deploy a scaled object with the `openstack-swift` scale trigger which uses `TriggerAuthentication` and the Password method from above.
```
Comment (tomkerkhove) [inline content/docs/2.2/scalers/openstack-metric.md:61]:
```suggestion
Here is an example of how to deploy a scaled object with the `openstack-metric` scale trigger which uses `TriggerAuthentication` and the Password method from above.
```

File: content/docs/2.2/scalers/openstack-metric.md [hunk 4]
```diff
@@ -0,0 +1,174 @@
...
+    aggregationMethod: "mean" #required
+    granularity: 300 #required (seconds)
+    threshold: 1250 #required
+    timeout: 30 #optional
+```
+
+> Please, always provide the protocol (http or https) when specifying URLs. This is needed due to Go URL parsing issues :sweat_smile:
+
+**Parameter list:**
+- `metricsURL` - The URL to query the metrics gnocchi API. It must contain the hostname, the gnocchi port, the API version, and the resource ID. The pattern is: `http://<host>:<swift_port>/<swift_version>/<resource_id>`
```
Comment (tomkerkhove) [inline content/docs/2.2/scalers/openstack-metric.md:33]: Is `gnocchi` required here?

File: content/docs/2.2/scalers/openstack-metric.md (hunk 1 above)
Comment (Rodolfodc) [reply content/docs/2.2/scalers/openstack-metric.md:13]: Hi, thanks for noticing it! the lines 1 to 5 are actually 'dead code' I should have deleted it. I will implement all changes you pointed and improve the explanation about gnocchi integration with openstack, thanks for the review.

Comment (Rodolfodc) [reply content/docs/2.2/scalers/openstack-metric.md]: Hi, I updated the file with the requested changes!

File: content/docs/2.2/scalers/openstack-metric.md (hunk 2 above)
Comment (Rodolfodc) [reply content/docs/2.2/scalers/openstack-metric.md]: Thanks, I have updated it.

File: content/docs/2.2/scalers/openstack-metric.md (hunk 3 above)
Comment (Rodolfodc) [reply content/docs/2.2/scalers/openstack-metric.md]: thanks, I updated it.

File: content/docs/2.2/scalers/openstack-metric.md (hunk 4 above)
Comment (Rodolfodc) [reply content/docs/2.2/scalers/openstack-metric.md]: Hi, I updated the docs about gnocchi. Actually gnocch is not required, its an openstack embedded time series database aimed to store and make it ease to request metrics which provides an API with an standard, the version here is linked with gnocchi api version, but it depends on openstack configurations and what api version it is providing.

File: content/docs/2.2/scalers/openstack-metric.md [hunk 5]
```diff
@@ -0,0 +1,170 @@
...
+title = "Openstack Metric"
+layout = "scaler"
+availability = "v2.3+"
+maintainer = "Community"
+description = "Scale applications based on the metrics collected from an specific resource."
+go_file = "openstack_metric_scaler"
++++
+
+### Trigger Specification
+This specification describes the `openstack-metric` trigger for OpenStack metrics. The Openstack metric API follows [gnocchi](https://gnocchi.xyz/) API. Attempt to the fact that gnocchi api is an open-source time series database embedded in openstack system and every parameter on Openstack Metric API follows it's patterns but you don't need to reference anything from gnocchi. It scales based on an spcific measeure from a given resource metric.
```
Comment (tomkerkhove) [inline content/docs/2.2/scalers/openstack-metric.md:11]:
```suggestion
This specification describes the `openstack-metric` trigger for OpenStack metrics.

> The Openstack metric API follows [gnocchi](https://gnocchi.xyz/) API. Attempt to the fact that gnocchi API is an open-source time series database embedded in openstack system and every parameter on Openstack Metric API follows its patterns but you don't need to reference anything from gnocchi. It scales based on a specific measure from a given resource metric.
```

File: content/docs/2.2/scalers/openstack-metric.md [hunk 6]
```diff
@@ -0,0 +1,170 @@
...
+  metadata:
+    metricsURL: http://localhost:8041/v1/metric #required
+    metricID: 003bb589-166d-439d-8c31-cbf098d863de #required
+    aggregationMethod: "mean" #required
+    granularity: 300 #required (seconds)
+    threshold: 1250 #required
+    timeout: 30 #optional
+```
+
+> Please, always provide the protocol (http or https) when specifying URLs. This is needed due to Go URL parsing issues :sweat_smile:
```
Comment (tomkerkhove) [inline content/docs/2.2/scalers/openstack-metric.md:26]:
```suggestion
> Protocol (http or https) should always be provided when specifying URLs
```

File: content/docs/2.2/scalers/openstack-metric.md [hunk 7]
```diff
@@ -0,0 +1,170 @@
++++
+title = "Openstack Metric"
+layout = "scaler"
+availability = "v2.3+"
+maintainer = "Community"
+description = "Scale applications based on the metrics collected from an specific resource."
```
Comment (tomkerkhove) [inline content/docs/2.2/scalers/openstack-metric.md:6]: Can you provide a more clear description, please?

**Review by tomkerkhove — APPROVED** [review]
LGTM, just waiting for the feature to merge

File: content/docs/2.2/scalers/openstack-metric.md (hunk 7 above)
Comment (Rodolfodc) [reply content/docs/2.2/scalers/openstack-metric.md:6]: sure!

Comment (Rodolfodc) [reply content/docs/2.2/scalers/openstack-metric.md]: Hi, I updated this description and linked the scaler PR to this doc PR.

File: content/docs/2.2/scalers/openstack-metric.md [hunk 8]
```diff
@@ -0,0 +1,172 @@
++++
+title = "Openstack Metric"
```
Comment (zroubalik) [inline content/docs/2.2/scalers/openstack-metric.md:2]:
```suggestion
title = "OpenStack Metric"
```

File: content/docs/2.2/scalers/openstack-metric.md [hunk 9]
```diff
@@ -0,0 +1,172 @@
...
+layout = "scaler"
+availability = "v2.3+"
+maintainer = "Community"
+description = "Scale applications based on a threshold reached by a specific measure from Openstack Metric API."
```
Comment (zroubalik) [inline content/docs/2.2/scalers/openstack-metric.md:6]:
```suggestion
description = "Scale applications based on a threshold reached by a specific measure from OpenStack Metric API."
```

File: content/docs/2.2/scalers/openstack-metric.md [hunk 10]
```diff
@@ -0,0 +1,172 @@
...
+    timeout: 30 #optional
+```
+
+> Protocol (http or https) should always be provided when specifying URLs
+
+**Parameter list:**
+- `metricsURL` - The URL to check for the metrics API, based. It must contain the hostname, the metric port, the API version, and the resource ID. The pattern is: `http://<host>:<metric_port>/<openstack_metric_api_version>/<resource_id>/metric` 
+- `metricID` - The Id of the intendend metric
+- `aggregationMethod` - The aggregation method that will be used to calculate metrics, it must follows the configured possible metrics derived from gnocchi API like: `mean`, `min`, `max`, `std`, `sum`, `count`, the complete aggregation methods list can be found [here](https://gnocchi.xyz/rest.html#archive-policy).
+- `granularity` - The configured granularity from metric collection in seconds. it must follow the same value configured in openstack, but it must be coutned in seconds. Sample: If you have a 5 minutes time window granularity defined, so you must input a value of 300 seconds (5*60).
```
Comment (zroubalik) [inline content/docs/2.2/scalers/openstack-metric.md:34]:
```suggestion
- `granularity` - The configured granularity from metric collection in seconds. it must follow the same value configured in OpenStack, but it must be coutned in seconds. Sample: If you have a 5 minutes time window granularity defined, so you must input a value of 300 seconds (5*60).
```

File: content/docs/2.2/scalers/openstack-metric.md [hunk 11]
```diff
@@ -0,0 +1,172 @@
...
+availability = "v2.3+"
+maintainer = "Community"
+description = "Scale applications based on a threshold reached by a specific measure from Openstack Metric API."
+go_file = "openstack_metric_scaler"
++++
+
+### Trigger Specification
+This specification describes the `openstack-metric` trigger for OpenStack metrics.
+
+> The Openstack metric API follows [gnocchi](https://gnocchi.xyz/) API. Attempt to the fact that gnocchi API is an open-source time series database embedded in openstack system and every parameter on Openstack Metric API follows its patterns but you don't need to reference anything from gnocchi. It scales based on a specific measure from a given resource metric.
```
Comment (zroubalik) [inline content/docs/2.2/scalers/openstack-metric.md:13]:
```suggestion
> The OpenStack metric API follows [Gnocchi](https://gnocchi.xyz/) API. Attempt to the fact that Gnocchi API is an open-source time series database embedded in OpenStack system and every parameter on OpenStack Metric API follows its patterns but you don't need to reference anything from Gnocchi. It scales based on a specific measure from a given resource metric.
```

**Review by zroubalik — CHANGES_REQUESTED** [review]
Could you move the content to `content/docs/2.3/` ?

File: content/docs/2.2/scalers/openstack-metric.md [hunk 12]
```diff
@@ -0,0 +1,172 @@
...
+It's highly recommended to check [gnocchi](https://gnocchi.xyz/) docs.
```
Comment (zroubalik) [inline content/docs/2.2/scalers/openstack-metric.md:14]:
```suggestion
It's highly recommended to check [Gnocchi](https://gnocchi.xyz/) docs.
```

**Comment by Rodolfodc** [comment]
Hi, I made a mistake while rebasing and I found out that it would be easier if I created a new branch! I closed this and opened this [new one](https://github.com/kedacore/keda-docs/pull/409). Sorry for any inconvenience

**Pull request closed without merging**